classes/actions.php

```diff
@@ -276,6 +312,26 @@ public static function duplicate_to_course(array $modules, int $targetcourseid,
                 moveto_module($targetmodinfo->get_cm($modid), $targetsection);
             }
         }
+        $event = \block_massaction\event\course_modules_duplicated::create([
+            'context' => \context_course::instance($targetcourseid),
```

Comment:
Why do you think we should trigger all those events in a context of a target course and not a source course?

Reply:
I changed to `$sourcecourseid` instead of `$targecourseid`